store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -4,49 +4,71 @@
 import com.zimbra.common.util.ZimbraLog;
 import com.zimbra.cs.event.Event;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Iterator;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
 
 public class EventLogger {
-    private static EventLogger eventLogger = null;
-    private final List<EventLogHandler> eventLogHandlers = new ArrayList<>();
+    private static final EventLogger eventLogger = new EventLogger();
+    private static final CopyOnWriteArrayList<EventLogHandler> eventLogHandlers = new CopyOnWriteArrayList<>();
+    private static final LinkedBlockingQueue<Event> eventQueue = new LinkedBlockingQueue<>();
+    private ExecutorService executorService = Executors.newFixedThreadPool(2);
 
-    private EventLogger() {}
+    private EventLogger() {
+        executorService.submit(new EventNotifier());
```

Comment:
Won't this only result in one executor thread polling the queue, even though the thread pool can handle 2?